duckduckgo/iOS
# BSK changes for NetP iOS Geoswitching

Task/Issue URL: https://app.asana.com/0/0/1205919928474411/f
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/557

**Description**:

This introduces the cross-platform changes for Network Protection Geoswitching. The general idea is:
- We make a request to the `locations` endpoint to get a list of server locations
- The user can select one of these locations
- We can use either a country or a country + city pairing to select a specific server location
- If none are selected, `register` will just return the nearest for their IP

**Steps to test this PR**:
1. Ensure you are an internal user (Settings -> Debug)
2. Go to Settings -> Network Protection -> VPN Settings
3. Check that the `Preferred Location` is nearest available
4. Tap through to see the list of locations
5. Select one (it’s a bare minimum UI for now so you won’t see any checkmark etc)
6. Go back to the main Settings -> Network Protection screen
7. **Observe that your server is the location you chose**
8. Repeat steps 2 and 4
9. Select `Nearest available`
10. Repeat step 6
11. **Observer that your server is the one nearese to you (the one you’ve had before this feature)**


**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
